app/Sources/TuistOnboarding/AppleSignInDelegate.swift

```diff
@@ -0,0 +1,37 @@
+import AuthenticationServices
+import Foundation
+import TuistAuthentication
+import TuistErrorHandling
+
+final class AppleSignInDelegate: NSObject, ASAuthorizationControllerDelegate,
+    ASAuthorizationControllerPresentationContextProviding
+{
+    let authenticationService: AuthenticationService
+    let errorHandling: ErrorHandling
```

Comment:
I'm a bit confused by the `ErrorHandling` pattern. There's a view modifier, which has an instance of `ErrorHandling` and subscribes to changes in `currentAlert` to show the alert if a new error has been fired, but in this delegate you are just creating another instance of the handler. Am I getting it right that the errors handled here won't be presented?

Reply:
I'm not creating another instance, I'm just storing a reference to it as we don't have access here to environment objects.